doc/cvode/guide/source/Usage/index.rst

```diff
@@ -1310,7 +1314,22 @@ difference approximation or a call to the :ref:`user-supplied Jacobian function
      * ``CVLS_ILL_INPUT`` -- The frequency ``msbj`` is negative.
 
    **Notes:**
-      The Jacobian update frequency is only checked within calls to the  linear solver setup routine, as such values of ``msbj`` < ``msbp`` will  result in recomputing the Jacobian every ``msbp`` steps. See :c:func:`CVodeSetLSetupFrequency` for setting the linear solver setup frequency  ``msbp``.  If ``msbj = 0``, the default value of 51 will be used. Otherwise an error is returned. This function must be called after the CVLS linear solver interface has been initialized through a call to :c:func:`CVodeSetLinearSolver`.
+      If ``nstlj`` is the step number at which the Jacobian information was
+      lasted updated and ``nst`` is the current step number,
+      ``nst - nstlj >= msbj`` indicates the Jacobian information will be updated
+      during the next linear solver setup call.
+
+      As the Jacobian update frequency is only checked *within* calls to the
+      linear solver setup routine, Jacobian information may be more than
+      ``msbj`` steps old when updated depending on when a linear solver setup
+      call occurs. See :numref:`CVODE.Mathematics.nls` for more information on
+      when linear solver setups are preformed.
+
+      If ``msbj = 0``, the default value of 51 will be used. Otherwise an error
```

Comment:
Is this really 51?

Reply:
Yes, originally the `msbj` value was 50 and the comparison to the last setup count used `>`. So a value of `0` would corresponding to updating every step (assuming the setup is called every step). `msbp` used a `>=` for the comparison and `1` would corresponded to setting up every setup. A while back we decided to make these more consistent so `1` would be setup every step. This required updating the default `msbj` value from `50` to `51` to keep the prior behavior.